src/wallet/rpcwallet.cpp

```diff
@@ -2298,6 +2298,17 @@ UniValue walletpassphrase(const JSONRPCRequest& request)
     // Alternately, find a way to make request.params[0] mlock()'d to begin with.
     strWalletPass = request.params[0].get_str().c_str();
 
+    // Get the timeout
+    int64_t nSleepTime = request.params[1].get_int64();
+    // Timeout cannot be negative, otherwise it will relock immediately
+    if (nSleepTime < 0) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Timeout cannot be negative.");
+    }
+    // Clamp timeout to 2^31 seconds
+    if (nSleepTime > ((int64_t)1 << 31) - 1) {
+        nSleepTime = ((int64_t)1 << 31) - 1;
```

Comment:
You could use the named constexpr `std::numeric_limits<int32_t>::max()`.

Also, how does that take into account your observation that "the time_t type that is used for the timer can be a 32 bit signed integer on some systems". (Considering that nSleepTime is multiplied by 1000 somewhere down in the call stack)

Reply:
> You could use the named constexpr `std::numeric_limits<int32_t>::max()`.

I'll try that now.

> Also, how does that take into account your observation that "the time_t type that is used for the timer can be a 32 bit signed integer on some systems". (Considering that nSleepTime is multiplied by 1000 somewhere down in the call stack)

It's multiplied by 1000 and then divided by 1000 before going into the `time_t`.

Reply:
Ah, ok. But since nSleepTime represents a delta, that is added onto `GetTime()`, capping it at the max of int32_t is not enough?

Reply:
The timer itself uses the delta. The only thing that's added is `wallet->nRelockTime` which is only used for display in `getwalletinfo`.